main/src/ca/uwaterloo/flix/language/ast/Ast.scala

```diff
@@ -387,4 +387,10 @@ object Ast {
     */
   case class IntroducedBy(clazz: java.lang.Class[_]) extends scala.annotation.StaticAnnotation
 
+
+  /**
+    * Represents that an instance on type `tpe` has the type constraints `tconstrs`.
+    */
+  case class Instance(tpe: Type, tconstrs: List[TypedAst.TypeConstraint])
```

Comment:
Wait.. If this is *only* used in phases that deal with TypedAst then it should be in TypedAst.

Reply:
TypeConstraint actually ends up being used outside of Typed AST, so I think that should be the one to move

Reply:
OK; do you want to make these two changes and then I can merge?